libs/execution_context/ebpf_core.c

```diff
@@ -230,7 +230,7 @@ ebpf_core_load_code(
     EBPF_LOG_ENTRY();
     ebpf_result_t retval;
     ebpf_program_t* program = NULL;
-    retval = ebpf_reference_object_by_handle(program_handle, EBPF_OBJECT_PROGRAM, (ebpf_core_object_t**)&program);
+    retval = ebpf_object_reference_by_handle(program_handle, EBPF_OBJECT_PROGRAM, (ebpf_core_object_t**)&program);
```

Comment:
Can we keep the original name, to minimize diffs?  Or is there a real reason to rename?

Reply:
Earlier, there was only one function `ebpf_reference_object_by_handle` which was present in ebpf_handle.c and this function was used everywhere because there was only one object type which was used to create handles (ebpf_core_object_t). With this PR, now handles can be created for ebpf_native_module_t objects also. 

Hence a new wrapper function `ebpf_object_reference_by_handle` is added in ebpf_object.c which should be used to ref the objects derived from ebpf_core_object_t. All the functions in ebpf_object.c have a prefix "ebpf_object_", so the new added function also follows the same pattern.